### ISSUE TITLE
Make it clearer that the first login to a server is also a registration

### ISSUE THREAD

**Issue by MarkTraceur** [minetest/minetest#258]
Just talked with a new player who was confused by the login interface--maybe have an intermediate step that says "successfully registered with server: username (whatever), password ***\* (don't actually show anything meaningful for the password).

Thanks!

**Comment by nerzhul** [comment]
Hi, thanks for your suggestion, but a mod could do this

**Comment by ShadowNinja** [comment]
@nerzhul Not really.  There should be, eg, a retype password field on registration so you don't mistype your password.  Mods can only send a chat message to the effect of "You just registered," which isn't very usefull.

**Comment by nerzhul** [comment]
Okay, this could be better, you're right

**Comment by Wuzzy2** [comment]
Because this is still relevant, I am re-posting this from #1043:

lightonflux said:
> It is very confusing to new users when a server requires a password, just look in the forum or chats. That is because most people expect that the password is for accessing the server, not to "secure" their username.

I said:
> Maybe the whole UI around this password thing must be rethought. It is bad for several reasons:
> 
> * What lightonflux said
> * On first-time login, it *silently* creates account and protects it with password. This is very bad. It can cost a newbie several “burned” accounts until finally realizing that, since many servers (sadly) force users to enter passwords (e.g. user might just enter any random string just to be allowed into the server, later forgetting the password because of not realizing it was an account creation). It also took me a while until I finally “got” how it works. It's frustrating when you lose a good user name because of that.
> * On the other hand, there is no obvious way to know whether an account creation *failed* because the server is down. Maybe the packet arrived on the server and the account was created but then the connection broke down at this exact moment
> * No visible difference beween account *creation* and a *normal login*. It all happens “magically” and silently without the user being noticed at all
> * No requirement to enter a password twice on account creation. If you screwed up your password in the first login (aka the silent magic account creation), you have burned just another account
> * No way to know whether you are forced to enter a password or not until you try. Very annoying if you just want to quickly peek into a server (maybe this forced password thing should be dropped, but thats a different issue)
> 
> If you compare Minetest's login and account system to the account system to basically *every website* in the web, you will notice:
> * You will rarely see a website where the login dialog is the same as the register dialog. Login and registration are almost always two different things (from the user perspective)
> * Almost every registration dialog has two password fieds to avoid screwing up your account very early
> * Also, almost every website notifies the user right after account creation
> 
> All this is pretty basic stuff. There is zero reason for Minetest not to follow this de facto standard.
> 
> Well, so more thoughtwork needs to go into the UI, but I am myself not in the right mood for this either (sorry :-( ).

**Comment by paramat** [comment]
Wasn't addressed by #6849 which apparently asks for confirmation after registering.

**Comment by Ezhh** [comment]
Can this please be reopened and the title changed back? The actual issue here is about there being no easy way to see how to register. The password confirmation was one suggested way of fixing that (and I think is a separate issue entirely, and is not necessarily the right way to fix it).

**Comment by paramat** [comment]
Yes ok.

**Comment by orwell96** [comment]
This can be closed, I think

**Comment by rubenwardy** [comment]
The OP was implemented by the confirmation dialog - despite what the commenters say

Further improvements are captured by #8138
